### ISSUE TITLE
Browser Aptitude: React Route changes not noticed by `listenNavigation`

### ISSUE THREAD

**Issue by coltonbrugger** [open-olive/loop-development-kit#488]
Attempting to listen for navigation within a React application does not trigger each time a link within the application is clicked/accessed.

To reproduce:

* Run the Browser aptitude for a site using React routes (e.g. https://reactrouter.com/)
* Upon first loading a page from that site, you'll get a listener fire with the page URL
* Upon navigation using routed links within that React app, the aptitude will not fire.
* If at a different location and you hit refresh, `listenNavigation` will fire again.

We simply must to be able to see these routed/history transitions.

**Comment by iangallOlive** [comment]
Hey Colton! One thing I want to check on real fast - what permissions are you providing for the `browser` aptitude in your `package.json`?

**Comment by coltonbrugger** [comment]
```
"permissions": {
  "browser": {
    "urlDomains": [
      {
        "value": "*.thedomain.com/*"
      }
    ]
  },
}
```

**Comment by iangallOlive** [comment]
@coltonbrugger Gotcha, thanks for confirming. Just wanted to make sure you were using the wildcard(s).

From what we can tell, the current `listenNavigation` implementation only fires for true navigation events, whereas single page apps (like using React Router) only _manipulate_ the browser's history. From what I can tell, React Router uses the [history](https://www.npmjs.com/package/history) package, which uses with the browser's [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API), which _seems_ to confirm that thought.

Nonetheless, we completely agree that the `browser` aptitude's `listenNavigation` function should be able to support navigation within single page apps, and we will look into this. I'll create a card to track investigation/progress on a solution.

**Comment by MaggieMaSS** [comment]
Thanks for the feedback! This feature has been released on Olive Helps 0.32.2 :)
